src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -0,0 +1,550 @@
+
+    failed_names = list()
+
+    counter = 1
+    for assembly_name in assembly_files:
+
+        if counter % 5000 == 0:
+            print(f"{round(counter / len(assembly_files), 4) * 100}% finished at {datetime.now()}")
+
+        assembly_path = os.path.join(JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER, DATA_DIR, assembly_name)
+        try:
+            _upload_assembly_to_workspace(conf, workspace_name, assembly_path, assembly_name)
+        except Exception as e:
+            print(e)
+            failed_names.append(assembly_name)
+
+        counter += 1
+
+    if failed_names:
+        print(f'Failed to upload {failed_names}')
+
+    return failed_names
+
+
+def create_entries_in_ws_and_softlinks_in_csd(
+        csd_upload: str,
+        output_dir: str,
+        assembly_names: list[str],
+        all_assemblies: dict[str, str],
+) -> None:
+    """
+    Create standard entries in workspace and softlinks in collectionssource directory.
+
+    csd_upload: directory of assemblies that have been uploaded to workspace
+    output_dir: output directory to create entries in workspace
+    assembly_names: a list of assembly names that exist in target workspace
+    all_assemblies: a dictionary of assembly name to file path
+    """
+    assembly_ids = create_entries_in_sd_workspace(assembly_names, all_assemblies, output_dir)
+    loader_helper.create_softlinks_in_csd(csd_upload, output_dir, assembly_ids)
+
+
+def main():
+    parser = _get_parser()
+    args = parser.parse_args()
+
+    workspace_id = args.workspace_id
+    kbase_collection = getattr(args, loader_common_names.KBASE_COLLECTION_ARG_NAME)
+    source_version = getattr(args, loader_common_names.SOURCE_VER_ARG_NAME)
+    root_dir = args.root_dir
+    token_filepath = args.token_filepath
+    upload_file_ext = args.upload_file_ext
+    keep_job_dir = args.keep_job_dir
+    overwrite = args.overwrite
+
+    env = args.env
+    kb_base_url = loader_common_names.KB_BASE_URL_MAP[env]
+
+    if workspace_id <= 0:
+        parser.error(f"workspace_id needs to be > 0")
+
+    uid = os.getuid()
+    username = os.getlogin()
+
+    job_dir = loader_helper.make_job_dir(
+        root_dir, loader_common_names.SDK_JOB_DIR, username
+    )
+    csd = loader_helper.make_collection_source_dir(
+        root_dir, loader_common_names.DEFAULT_ENV, kbase_collection, source_version
+    )
+    csd_upload = loader_helper.make_collection_source_dir(
+        root_dir, loader_common_names.DEFAULT_ENV, kbase_collection, source_version, True
+    )
+    output_dir = loader_helper.make_output_dir(
+        root_dir,
+        loader_common_names.SOURCE_DATA_DIR,
+        loader_common_names.WS,
+        env,
+        workspace_id,
+    )
+
+    proc = None
+    conf = None
+
+    try:
+        # start podman service
+        proc = loader_helper.start_podman_service(uid)
+    except Exception as e:
+        raise Exception("Podman service failed to start") from e
```

Comment:
it will catch the ValueError raised by [start_podman_service](https://github.com/kbase/collections/blob/35d6309a2c4b082acf855d087264e5fdba188bc9/src/loaders/common/loader_helper.py#L162)

Reply:
Right, but for what purpose? It's adding the string around the service failing, but why not add that string to the original exception and just not catch it here and let it halt the script?

Reply:
Updated!

Reply:
I don't think there's any need for the `except` or `else` blocks at this point. The exception can just be allowed to bubble up the call stack

Reply:
If we get rid of except and else, the failing service would halt the script. Isn't that something we try to avoid?

Reply:
That's the way it worked before right?
* podman fails
* exception thrown from helper method
* exception caught in main method
* exception rethrown, script halts

This is no different except the exception isn't caught and rethrown.

Reply:
Updated!

Reply:
:+1: